client/client_test.go

```diff
@@ -33,6 +34,91 @@ func TestAddGetLeaf(t *testing.T) {
 	// TODO: Build a GetLeaf method and test a full get/set cycle.
 }
 
+func TestGetByIndex(t *testing.T) {
+	env, err := integration.NewLogEnv(context.Background(), 0, "TestGetByIndex")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer env.Close()
+	logID, err := env.CreateLog()
+	if err != nil {
+		t.Fatalf("Failed to create log: %v", err)
+	}
+
+	cli := trillian.NewTrillianLogClient(env.ClientConn)
+	client := New(logID, cli, testonly.Hasher, env.PublicKey)
+	// Add a few test leaves.
+	leafData := [][]byte{
+		[]byte("A"),
+		[]byte("B"),
+		[]byte("C"),
+	}
+
+	// TODO(gdbelvin): Replace with batch API.
+	// TODO(gdbelvin): Replace with AddSequenedLeaves API.
+	for _, l := range leafData {
+		ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(timeout))
+		defer cancel()
+		if err, want := client.AddLeaf(ctx, l), codes.DeadlineExceeded; grpc.Code(err) != want {
+			t.Fatalf("AddLeaf(%v): %v, want, %v", l, err, want)
+		}
+		env.Sequencer.OperationLoop() // Sequence the new leaves in-order.
+	}
+
+	for i, l := range leafData {
+		leaf, err := client.GetByIndex(context.Background(), int64(i))
+		if err != nil {
+			t.Errorf("Failed to GetByIndex(%v): %v", i, err)
+		}
+		if got, want := leaf.LeafValue, l; !bytes.Equal(got, want) {
+			t.Errorf("GetByIndex(%v) = %v, want %v", i, got, want)
+		}
+	}
+}
+
+func TestListByIndex(t *testing.T) {
+	env, err := integration.NewLogEnv(context.Background(), 0, "TestGetByIndex")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer env.Close()
+	logID, err := env.CreateLog()
+	if err != nil {
+		t.Fatalf("Failed to create log: %v", err)
+	}
+
+	cli := trillian.NewTrillianLogClient(env.ClientConn)
+	client := New(logID, cli, testonly.Hasher, env.PublicKey)
+	// Add a few test leaves.
+	leafData := [][]byte{
+		[]byte("A"),
+		[]byte("B"),
+		[]byte("C"),
+	}
+
+	// TODO(gdbelvin): Replace with batch API.
+	// TODO(gdbelvin): Replace with AddSequenedLeaves API.
+	for _, l := range leafData {
+		ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(timeout))
+		defer cancel()
+		if err, want := client.AddLeaf(ctx, l), codes.DeadlineExceeded; grpc.Code(err) != want {
+			t.Fatalf("AddLeaf(%v): %v, want, %v", l, err, want)
+		}
+		env.Sequencer.OperationLoop() // Sequence the new leaves in-order.
+	}
+
+	// Fetch leaves.
+	leaves, err := client.ListByIndex(context.Background(), 0, 3)
```

Comment:
(already discussed, commenting for future reference) Test setup is pretty much identical. Suggestions are to refactor setup to a 3rd function or maybe merge the tests.